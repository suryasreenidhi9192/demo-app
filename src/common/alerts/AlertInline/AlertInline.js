import React from 'react'
import PropTypes from 'prop-types';

import {
    SvgNotificationSuccessLight,
    SvgNotificationInfoLight,
    SvgNotificationWarningLight,
    SvgNotificationErrorLight
} from '../../icons'

import './AlertInline.scss'

/**
 * @public
 * Creates an inline styled block for important messages for the user
 * @param  {String}     options.type        Error info etc. Used to toggle style of Alert
 * @param  {Boolean}    options.showAlert   Flag used to toggle visibility of Alert
 * @param  {Node}       options.children    React Component or string to render in the Alert body
*/
const AlertInline = ({ type = 'error', showAlert = true, children }) => {

    const Icon = (() => {

        switch (type) {

            case 'success':
                return SvgNotificationSuccessLight

            case 'info':
                return SvgNotificationInfoLight

            case 'warning':
                return SvgNotificationWarningLight

            default:
                return SvgNotificationErrorLight

        }

    })()

    return (

        showAlert && (

            <div className={`alert-inline ${type}`} role="alert">
                <Icon className="alert-icon" />
                <div className="message">{ children }</div>
            </div>

        )

    ) || null

}

AlertInline.propTypes = {
    /** React Component or string to render in the body */
    children: PropTypes.node,
    /** Error info etc. Used to toggle style of Alert */
    type: PropTypes.string
};

export default AlertInline
