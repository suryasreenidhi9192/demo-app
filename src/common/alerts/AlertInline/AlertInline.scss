@import "../../shared/css/variables";

.alert-inline {
    display: flex;
    align-items: center;
    margin: 16px 0;
    width: auto;
    padding: 16px;
    background-color: var(--content-background);
    border: 1px solid;
    border-radius: 6px;

    svg {
        flex: 0 0 24px;
        width: 24px;
        height: 24px;
    }

    &.success { border-color: var(--utility-green); }
    &.error { border-color: var(--utility-red); }
    &.warning { border-color: var(--utility-amber); }
    &.info { border-color: var(--utility-blue); }

    .message {
        flex: 1 1 auto;
        font-family: $lt;
        font-size: 16px;
        display: inline-block;
        margin: 0 16px;
    }

}

