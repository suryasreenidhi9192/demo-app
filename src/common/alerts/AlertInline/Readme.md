### Neutral Example
```jsx
<AlertInline type="neutral">
    It's a simple lift and shift job can I just chime in on that one low engagement nor drill down, so UI, or hammer out, but turd polishing.
</AlertInline>
```

### Info Example
```jsx
<AlertInline type="info">
     What's the status on the deliverables for eow? thought shower, yet parallel path.
</AlertInline>
```

### Empathy Example
```jsx
<AlertInline type="empathy">
    We need to make the new version clean and sexy first-order optimal strategies.
</AlertInline>
```

### Success Example
```jsx
<AlertInline type="success">
    It's a simple lift and shift job can I just chime in on that one low engagement nor drill down, so UI, or hammer out, but turd polishing.
</AlertInline>
```

### Warning Example
```jsx
<AlertInline type="warning">
    Pipeline granularity we just need to put these last issues to bed, i don't want to drain the whole swamp, i just want to shoot some alligators or circle back we need to button up our approach.
</AlertInline>
```

### Error Example
```jsx
<AlertInline type="error">
    We need to make the new version clean and sexy first-order optimal strategies.
</AlertInline>
```
