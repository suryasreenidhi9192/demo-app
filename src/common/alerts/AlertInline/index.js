import AlertInline from './AlertInline'

export default AlertInline
