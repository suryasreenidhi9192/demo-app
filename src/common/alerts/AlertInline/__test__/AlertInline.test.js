import React from 'react'
import { mount } from 'enzyme';
import { matchSnapshot } from '../../../../../__test__/__utils__'

import AlertInline from '../'

const props = {
    children: (<b>This is some copy for the body of the AlertInline</b>)
}

const errorProps = {
    ...props,
    type: 'error'
}

const successProps = {
    ...props,
    type: 'success'
}

const warningProps = {
    ...props,
    type: 'warning'
}

const infoProps = {
    ...props,
    type: 'info'
}

const checkDefaultType = () => {
    expect(mount(
        <AlertInline { ...props } />
    ).find('.error').exists()).toBe(true)
}

describe('AlertInline:.......................SnapShot Testing:', () => {
    test(
        'Error AlertInline Matches jest snapshot',
        matchSnapshot(AlertInline, errorProps)
    )
    test(
        'Success AlertInline Matches jest snapshot',
        matchSnapshot(AlertInline, successProps)
    )
    test(
        'Warning AlertInline Matches jest snapshot',
        matchSnapshot(AlertInline, warningProps)
    )
    test(
        'Info AlertInline Matches jest snapshot',
        matchSnapshot(AlertInline, infoProps)
    )
})

describe('AlertInline:.......................Prop Rendering Logic:', () => {
    test(
        'Error AlertInline is the default value',
        checkDefaultType
    )
})
