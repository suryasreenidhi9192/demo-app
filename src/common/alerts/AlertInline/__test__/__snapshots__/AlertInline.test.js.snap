// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertInline:.......................SnapShot Testing: Error AlertInline Matches jest snapshot 1`] = `
<div
  className="alert-inline error"
  role="alert"
>
  <svg
    className="alert-icon"
    viewBox="0 0 32 32"
  >
    <title>
      
        Error Notification
    
    </title>
    <g
      fill="none"
      fillRule="evenodd"
    >
      <path
        className="Notification-Error-Light_svg__path-01"
        d="M21.744 29.867l8.123-8.123V10.256l-8.123-8.123H10.256l-8.123 8.123v11.488l8.123 8.123h11.488zM8.956 31.583l-8.331-8.33c-.4-.4-.625-.944-.625-1.51V10.257c0-.565.225-1.108.625-1.508L8.748.625c.4-.4.943-.625 1.508-.625h11.488c.566 0 1.109.225 1.509.625l8.122 8.123c.4.4.625.943.625 1.508v11.488c0 .565-.225 1.108-.625 1.508l-8.122 8.123c-.4.4-.943.625-1.509.625H9.962c-.377 0-.74-.15-1.006-.417z"
      />
      <path
        className="Notification-Error-Light_svg__path-02"
        d="M17.27 20.64a.825.825 0 01-.145.488.52.52 0 01-.171.16.356.356 0 01-.174.045h-1.56c-.133 0-.248-.068-.345-.205a.825.825 0 01-.145-.489l-.508-12.834c0-.188.049-.351.145-.488.097-.137.212-.206.345-.206h2.576c.133 0 .248.069.345.206.096.137.145.3.145.488L17.27 20.64zM16 25.6a1.422 1.422 0 110-2.844 1.422 1.422 0 010 2.844z"
      />
    </g>
  </svg>
  <div
    className="message"
  >
    <b>
      This is some copy for the body of the AlertInline
    </b>
  </div>
</div>
`;

exports[`AlertInline:.......................SnapShot Testing: Info AlertInline Matches jest snapshot 1`] = `
<div
  className="alert-inline info"
  role="alert"
>
  <svg
    className="alert-icon"
    viewBox="0 0 32 32"
  >
    <title>
      
        Information Notification
    
    </title>
    <g
      fill="none"
      fillRule="evenodd"
    >
      <path
        className="Notification-Info-Light_svg__path-01"
        d="M16 29.867c7.658 0 13.867-6.209 13.867-13.867 0-7.658-6.209-13.867-13.867-13.867C8.342 2.133 2.133 8.342 2.133 16c0 7.658 6.209 13.867 13.867 13.867zM16 32C7.163 32 0 24.837 0 16S7.163 0 16 0s16 7.163 16 16-7.163 16-16 16z"
      />
      <path
        className="Notification-Info-Light_svg__path-02"
        d="M16 11.022a1.6 1.6 0 100-3.2 1.6 1.6 0 000 3.2zm1.422 13.895a.71.71 0 01-.175.48.533.533 0 01-.416.203H15.17a.533.533 0 01-.416-.202.71.71 0 01-.175-.481v-11.79a.71.71 0 01.175-.48.533.533 0 01.416-.203h1.662c.16 0 .3.068.416.203a.71.71 0 01.175.48v11.79z"
      />
    </g>
  </svg>
  <div
    className="message"
  >
    <b>
      This is some copy for the body of the AlertInline
    </b>
  </div>
</div>
`;

exports[`AlertInline:.......................SnapShot Testing: Success AlertInline Matches jest snapshot 1`] = `
<div
  className="alert-inline success"
  role="alert"
>
  <svg
    className="alert-icon"
    viewBox="0 0 32 32"
  >
    <title>
      
        Success Notification
    
    </title>
    <g
      fill="none"
      fillRule="evenodd"
    >
      <path
        className="Notification-Success-Light_svg__path-01"
        d="M16 29.867c7.658 0 13.867-6.209 13.867-13.867 0-7.658-6.209-13.867-13.867-13.867C8.342 2.133 2.133 8.342 2.133 16c0 7.658 6.209 13.867 13.867 13.867zM16 32C7.163 32 0 24.837 0 16S7.163 0 16 0s16 7.163 16 16-7.163 16-16 16z"
      />
      <path
        className="Notification-Success-Light_svg__path-02"
        d="M12.936 19.61l9.71-9.984a1.422 1.422 0 112.039 1.983L12.993 23.63l-6.1-5.927a1.422 1.422 0 011.981-2.04l4.062 3.946z"
      />
    </g>
  </svg>
  <div
    className="message"
  >
    <b>
      This is some copy for the body of the AlertInline
    </b>
  </div>
</div>
`;

exports[`AlertInline:.......................SnapShot Testing: Warning AlertInline Matches jest snapshot 1`] = `
<div
  className="alert-inline warning"
  role="alert"
>
  <svg
    className="alert-icon"
    viewBox="0 0 32 32"
  >
    <title>
      
        Warning Notification
    
    </title>
    <g
      fill="none"
      fillRule="evenodd"
    >
      <path
        className="Notification-Warning-Light_svg__path-01"
        d="M16.968 2.767a1.084 1.084 0 00-1.938 0L2.282 28.264a1.084 1.084 0 00.97 1.569h25.496a1.084 1.084 0 00.97-1.569L16.967 2.767zm1.939-.97l12.749 25.498A3.251 3.251 0 0128.748 32H3.251a3.251 3.251 0 01-2.908-4.705L13.091 1.798a3.251 3.251 0 015.816 0z"
      />
      <path
        className="Notification-Warning-Light_svg__path-02"
        d="M17.27 22.417a.825.825 0 01-.145.488.52.52 0 01-.171.162.356.356 0 01-.174.044h-1.56c-.133 0-.248-.068-.345-.206a.825.825 0 01-.145-.488l-.508-12.123c0-.189.049-.351.145-.488.097-.137.212-.206.345-.206h2.576c.133 0 .248.069.345.206.096.137.145.3.145.488l-.508 12.123zM16 27.377a1.422 1.422 0 110-2.844 1.422 1.422 0 010 2.845z"
      />
    </g>
  </svg>
  <div
    className="message"
  >
    <b>
      This is some copy for the body of the AlertInline
    </b>
  </div>
</div>
`;
